### PR TITLE
Implemented Signup and Login page

### PR DESCRIPTION
**Pull request by ChangGittyHub**
Includes:
- Frontend Validation
- Correct routing from sign up to log in
- Option to sign in via Username or email address
- Password validation matches Chenyu's backend password validation

### CODE REVIEW

File: backend/.env.local [hunk 1]
```diff
@@ -0,0 +1 @@
+MONGODB_URI=mongodb+srv://rugatugtuck:x0wkDqgOvSiDtANx@peerprepcluster.gdf9nh3.mongodb.net/questions
```
Comment (PeanutButters93) [inline backend/.env.local]: Should gitignore this file for security reason.

File: frontend/.idea/checkstyle-idea.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (PeanutButters93) [inline frontend/.idea/checkstyle-idea.xml]: Should gitignore .idea folder. I think it contains user specific info, but I'm not sure

**Review by PeanutButters93 — COMMENTED** [review]
Pages work fine when tested, would recommend to add some files to .gitignore first

File: backend/.env.local (hunk 1 above)
Comment (zzthian) [reply backend/.env.local]: Yup please do so

**Review by zzthian — APPROVED** [review]
Works overall. LGTM.

**Pull request merged**